pkg/operator/admission/validation/pullsecret/http_server.go

```diff
@@ -0,0 +1,115 @@
+package pullsecret
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"embed"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/gofrs/uuid"
+	"github.com/sirupsen/logrus"
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	aroclient "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned"
+	"github.com/Azure/ARO-RP/pkg/util/deployer"
+	"github.com/Azure/ARO-RP/pkg/util/dynamichelper"
+)
+
+const rpImage = "RP_IMAGE"
+
+//go:embed manifests
+var assets embed.FS
+
+type deploymentData struct {
+	Image string
+}
+
+func Deploy(ctx context.Context, arocli aroclient.Interface, kubernetescli kubernetes.Interface, dh dynamichelper.Interface) error {
+	aroCluster, err := arocli.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	image, isSet := os.LookupEnv(rpImage)
+	if !isSet {
+		return fmt.Errorf("%s is not set", rpImage)
+	}
+
+	dep := deployer.NewDeployer(kubernetescli, dh, assets, "manifests")
+	return dep.CreateOrUpdate(context.Background(), aroCluster, deploymentData{Image: image})
+}
+
+func StartValidator(ctx context.Context, log *logrus.Entry, certPath, keyPath string) error {
+	c := ociRegClient{
+		httpClient: &http.Client{Timeout: time.Second * 3},
+		ctx:        ctx,
+		required:   map[string]bool{"quay.io": true},
+		log:        log,
+	}
+	http.HandleFunc("/", c.handleRequest)
+	return http.ListenAndServeTLS(":8080", certPath, keyPath, nil)
```

Comment:
this is not useful.

Reply:
this is just wrong Aldo, it makes the code more complicated to read because you have to go back and forth between different places in the code. You are just suggesting to make the code more complicated and less readable.